feature-find-meal/src/main/java/com/kinandcarta/feature/find/meal/adapter/MealsAdapter.kt

```diff
@@ -14,7 +14,7 @@ import com.kinandcarta.feature.find.meal.R
 import com.kinandcarta.feature.find.meal.viewmodel.Meals
 
 class MealsAdapter(
-    private var mealsList: List<Meal> = listOf(),
+    private var mealsList: MutableList<Meal> = mutableListOf(),
```

Comment:
Made it mutable so that we can update the item that changed on the reserve only, rather than having to refresh the whole list.  Which is a worse UX.

Reply:
Nice. I think we could also use [ListAdapter](https://developer.android.com/reference/androidx/recyclerview/widget/ListAdapter) to handle a lot of this for us but this works great as well

Reply:
Playing the other side here:
I'm a user that tried to reserve a meal - turns out my list is not up to date and that meal is gone so I get an alert (or toast) saying "Sorry that meal is unavailable"
and that item is updated to say as much.
I go to the next item nearby and tap to reserve, but this item is also not up to date.
I get the same alert and that one item updates to show it's unavailable.

Could be annoying?

Reply:
Another thing maybe to consider is that if the user is moving.
One items distance gets updated but the rest of the list doesn't

Reply:
Perhaps we need to consider how up to date the list is going to need to be to cover those scenarios?

Reply:
Not sure what you mean

Reply:
What I meant was, once we have a list, after a little bit of time it may be out of date, regarding the reservation status and distance.   So perhaps we might want to think about adding a refresh list reminder or timed / event driven refresh

Reply:
Yeah something to think on

Reply:
Have updated to use ListAdapter diffing, allowing the List to be updated as a whole.